File: Assets/AltUnityTester-BindingsAndExamples(can_be_deleted)/Bindings~/python/altunityrunner/commands/InputActions/scroll_mouse_and_wait.py
from altunityrunner.commands.command_returning_alt_elements import CommandReturningAltElements
from altunityrunner.commands.InputActions.scroll_mouse import ScrollMouse
import time
class ScrollMouseAndWait(CommandReturningAltElements):
    def __init__(self, socket,request_separator,request_end, speed, duration):
        super().__init__(socket,request_separator,request_end)
        self.speed=speed
        self.duration=duration
    
    def execute(self):
        data = ScrollMouse(self.socket,self.request_separator,self.request_end,self.speed, self.duration).execute()
        self.handle_errors(data)
        print('Wait for scroll mouse to finish')
        time.sleep(self.duration)
        action_in_progress = True
        while action_in_progress:
            action_finished = self.send_data(self.create_command('actionFinished'))
            self.handle_errors(action_finished)
            if action_finished is 'Yes':
                break
            elif action_finished != 'No':
                action_in_progress = False
        return self.handle_errors(data)

File: Assets/AltUnityTester/AltUnityDriver/Commands/InputActions/ScrollMouseAndWait.cs
public class ScrollMouseAndWait : AltBaseCommand
{
    float speed;
    float duration;
    public ScrollMouseAndWait(SocketSettings socketSettings, float speed, float duration) : base(socketSettings)
    {
        this.speed = speed;
        this.duration = duration;
    }
    public void Execute()
    {
        new ScrollMouse(SocketSettings,speed, duration).Execute();
        System.Threading.Thread.Sleep((int)duration * 1000);
        string data;
        do
        {
            Socket.Client.Send(toBytes(CreateCommand("actionFinished")));
            data = Recvall();
        } while (data == "No");
        if (data.Equals("Yes"))
            return;
        HandleErrors(data);
    }
}